kiota_abstractions/store/backed_model.py

```diff
@@ -1,17 +1,33 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation.  All Rights Reserved.
+# Licensed under the MIT License.
+# See License in the project root for license information.
+# ------------------------------------------------------------------------------
+
 from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Optional
 
 from .backing_store import BackingStore
 
 
-class BackedModel(ABC):
+@dataclass
+class BackedModel:
     """Defines the contracts for a model that is backed by a store.
     """
+    # Stores model information.
+    backing_store: BackingStore
+
+    def __setattr__(self, prop, val):
+        if not prop == "backing_store":
+            self.__backing_store.set(prop, val)
+            super().__setattr__(prop, self.__backing_store.get(prop))
+        super().__setattr__(prop, val)
```

Comment:
Actually if you look closely we are overriding the default implementation of `__setattr__` and calling the `set()` method of the backing store instead. Unless the property we are setting is the backing store, in that case we set the property as usual.

Reply:
I think there would be a bug in the scenario of fetching from the API, updating some attributes on an instance and PATCH-ing since during serialization, all model properties will return their initial value which was set during deserialization. The backing store get() needs to run at the point a model property's "get" happens.

Reply:
> Might we need a __getattr__ override which reads the property value from the backing store instead of "getters" returning the attribute value directly

Added here dfd385f912aadbce8b3587ad162515a5f2baeb40